addons/web/controllers/main.py

```diff
@@ -1457,6 +1457,22 @@ def content_common(self, xmlid=None, model='ir.attachment', id=None, field='data
             response.set_cookie('fileToken', token)
         return response
 
+    @http.route(['/web/assets/debug/<string:filename>',
+        '/web/assets/debug/<path:extra>/<string:filename>',
+        '/web/assets/<int:id>/<string:filename>',
+        '/web/assets/<int:id>-<string:unique>/<string:filename>',
+        '/web/assets/<int:id>-<string:unique>/<path:extra>/<string:filename>'], type='http', auth="public")
+    def content_assets(self, xmlid=None, model='ir.attachment', id=None, field='datas',
+                       filename=None, filename_field='name', unique=None, mimetype=None,
+                       download=None, data=None, token=None, access_token=None, **kw):
+
```

Comment:
Here, on the other hand, I would only accept the parameters that we *know* we'll be using. e.g. not `xmlid`, `model`, `field`, `filename_field`, `token`, `access_token`, `mimetype`, `download`, ...

Since this is a new route dedicated to assets there's no legacy to support for other URL patterns. If we accept more parameters now, it may make it harder to implement more asset-specific logic in the future.

It will also avoid potential security-related issues related to `model`, `token`, etc. in the future, as we don't want to open more doors here that strictly necessary.

`**kw` is kept to support extra params like `debug=xxx` without crashing, but we ignore query string params we don't *need* to accept.

```suggestion
    def content_assets(self, id=None, filename=None, unique=None, **kw):

```

Reply:
good time to make `extra` a named parameter too, no?